### PR TITLE
LPS-89637 ServiceBuilder without spring

### PR DESCRIPTION
**Pull request by Preston-Crary**
Haven't updated ServiceBuilder yet.  Just want to confirm with you the end goal before I start making changes to the ftl.

### CODE REVIEW

File: portal-kernel/src/com/liferay/portal/kernel/dao/jdbc/DataSourceProvider.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package com.liferay.portal.kernel.dao.jdbc;
+
+import javax.sql.DataSource;
+
+/**
+ * @author Preston Crary
+ */
+public interface DataSourceProvider {
+
+	public DataSource createDataSource();
```
Comment (shuyangzhou) [inline portal-kernel/src/com/liferay/portal/kernel/dao/jdbc/DataSourceProvider.java]: Rename it to getDataSource()

File: portal-kernel/src/com/liferay/portal/kernel/module/framework/service/IdentifiableOSGiService.java [hunk 2]
```diff
@@ -14,9 +14,12 @@
...
 
+import com.liferay.portal.kernel.transaction.Transactional;
+
 /**
  * @author Tina Tian
  */
+@Transactional(enabled = false)
 public interface IdentifiableOSGiService {
 
 	public String getOSGiServiceIdentifier();
```
Comment (shuyangzhou) [inline portal-kernel/src/com/liferay/portal/kernel/module/framework/service/IdentifiableOSGiService.java]: Move @Transactional(enabled = false) to method level

File: definitions/liferay-service-builder_7_2_0.dtd [hunk 3]
```diff
@@ -46,6 +46,15 @@ entities to prevent lost updates. The default value is false.
 	mvcc-enabled CDATA #IMPLIED
 >
 
+<!--
+The dependency-injector value specifies which type of dependency injection to
+use. Valid values are "spring" and "declarative-services" The default value is
```
Comment (shuyangzhou) [inline definitions/liferay-service-builder_7_2_0.dtd]: Just call "declarative-services" "ds"

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/base/BlogsEntryLocalServiceBaseImpl.java [hunk 4]
```diff
@@ -500,539 +502,26 @@ public BlogsEntry updateBlogsEntry(BlogsEntry blogsEntry) {
...
-	 * Sets the blogs stats user persistence.
-	 *
-	 * @param blogsStatsUserPersistence the blogs stats user persistence
-	 */
-	public void setBlogsStatsUserPersistence(
-		BlogsStatsUserPersistence blogsStatsUserPersistence) {
-		this.blogsStatsUserPersistence = blogsStatsUserPersistence;
+	@Override
+	public Class<?>[] getAopInterfaces() {
+		return new Class[] {
```
Comment (shuyangzhou) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/base/BlogsEntryLocalServiceBaseImpl.java]: new Class<?>[]

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/base/BlogsEntryLocalServiceBaseImpl.java [hunk 5]
```diff
@@ -500,539 +502,26 @@ public BlogsEntry updateBlogsEntry(BlogsEntry blogsEntry) {
...
 	}
 
-	public void afterPropertiesSet() {
-		persistedModelLocalServiceRegistry.register("com.liferay.blogs.model.BlogsEntry",
-			blogsEntryLocalService);
+	public void activate() {
 	}
 
-	public void destroy() {
+	public void deactivate() {
```
Comment (shuyangzhou) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/base/BlogsEntryLocalServiceBaseImpl.java]: This is going to be a problem if the subclass does not override it and apply Deactivate annotation.

**Comment by Preston-Crary** [comment]
Please see https://github.com/shuyangzhou/liferay-portal/pull/7369

**Pull request closed without merging**